doc/src/tutorial/codegen_explainatory.rst

```diff
@@ -0,0 +1,458 @@
+             )**2)/(hbar**(1.0d0/4.0d0)*pi**(1.0d0/4.0d0))
+    end do
+
+    end subroutine
+
+The argument ``verbose`` is boolean, optional and if True, autowrap 
+will not mute the command line backends. This can be helpful for debugging.
+
+The argument ``language`` and ``backend`` are used to change defaults: 'Fortran'
+and 'f2py' to 'C' and 'Cython'.
+The argument helpers is used to define auxillary expressions needed for the main 
+expression. If the main expression needs to call a specialized function it should 
+be put in the ``helpers`` iterable. Autowrap will then make sure that the
+compiled main expression can link to the helper routine. Items should
+be tuples with (<funtion_name>, <sympy_expression>, <arguments>). It is mandatory 
+to supply an argument sequence to helper routines.
+
+Another method available at the ``autowrap`` level is ``binary_function``. It returns 
+a sympy function. The advantage is that we can have very fast functions as compared
+to SymPy speeds. An illustration::
+
+    >>> from sympy.utilities.autowrap import binary_function
+    >>> from sympy import symbols
+    >>> from sympy.physics.hydrogen import R_nl
+    >>> a,r = symbols('a,r')
+    >>> psi_nl=R_nl(1,0,a,r)
+    >>> f=binary_function('f', psi_nl)
+    >>> f(a,r).evalf(3, subs={a: 1, r: 2})
+    0.766
+
+
+While NumPy operations are very efficient for vectorized data but they sometimes incur 
+unnecessary costs when chained together.  
+Fortunately, SymPy is able to generate efficient low-level C or Fortran code. 
+It can then depend on projects like Cython or f2py to compile and reconnect that 
+code back up to Python. Fortunately this process is well automated and a SymPy user 
+wishing to make use of this code generation should call the ufuncify function
+``ufuncify`` is the third method available with Autowrap module. 
+It basically implies 'Universal functions' and follows an ideology set by Numpy.
+Let us see an example::
+
+    >>> from sympy import *
+    >>> init_printing(use_latex='mathjax')
+    >>> from sympy.physics.hydrogen import R_nl
+    >>> x=symbols('x')
+    >>> expr=R_nl(3,1,x,6)
+    >>> expr
+                    -2⋅x
+    8⋅x⋅(-4⋅x + 4)⋅ℯ    
+    ────────────────────
+             3          
+
+The lambdify function translates SymPy expressions into Python functions, 
+leveraging a variety of numerical libraries.By default lambdify relies 
+on implementations in the ``math`` standard library. Naturally, Raw Python 
+is faster than Sympy. However it also supports ``mpmath`` and most notably, 
+``numpy``. Using the numpy library gives the generated function access to 
+powerful vectorized ufuncs that are backed by compiled C code.
+
+Let us compare the speeds::
+
+    >>> from sympy.utilities.autowrap import ufuncify
+    >>> from sympy.utilities.lambdify import lambdify
+    >>> fn_numpy=lambdify(x, expr,'numpy')
+    >>> fn_fortran=ufuncify([x], expr)
```

Comment:
Oh, yes! The default for ufuncify is numpy. I need to change this to add the backend=f2py arg.
Though, I can't guess why numpy is beating numpy.